src/mode/modeHandler.ts

```diff
@@ -129,50 +129,63 @@ export class VimSettings {
   useCtrlKeys     = false;
 }
 
+export enum SearchDirection {
+  Forward = 1,
+  Backward = -1
+}
+
 export class SearchState {
+  private static readonly MAX_SEARCH_RANGES = 1000;
+
   /**
    * Every range in the document that matches the search string.
    */
-  public matchRanges: vscode.Range[] = [];
+  private _matchRanges: vscode.Range[] = [];
+  public get matchRanges(): vscode.Range[] {
```

Comment:
I do like the idea of accessibilty restriction and per https://github.com/Microsoft/TypeScript/issues/2845, accessors for the same member name must specify the same accessibility, we can't have a public `getter` and a private `setter` at the same time, so accessing the private member to do modification is the best way.